src/as-validator.c

```diff
@@ -1257,10 +1264,69 @@ as_validator_check_screenshots (AsValidator *validator, xmlNode *node, AsCompone
 			if (iter2->type != XML_ELEMENT_NODE)
 				continue;
 
+			g_autofree gchar *screenshot_width = NULL;
+			g_autofree gchar *screenshot_height = NULL;
+
+			screenshot_width = as_xml_get_prop_value (iter2, "width");
+			if (screenshot_width != NULL && !as_str_verify_integer (screenshot_width, 1, G_MAXINT64))
+				as_validator_add_issue (validator, iter2,
+					"screenshot-invalid-width",
+					screenshot_width);
+
+			screenshot_height = as_xml_get_prop_value (iter2, "height");
+			if (screenshot_height != NULL && !as_str_verify_integer (screenshot_height, 1, G_MAXINT64))
+				as_validator_add_issue (validator, iter2,
+					"screenshot-invalid-height",
+					screenshot_height);
+
 			if (g_strcmp0 (node_name, "image") == 0) {
 				g_autofree gchar *image_url = as_strstripnl ((gchar*) xmlNodeGetContent (iter2));
+				g_autofree gchar *image_type = NULL;
 
 				image_found = TRUE;
+
+				image_type = as_xml_get_prop_value (iter2, "type");
+				if (image_type == NULL)
+					image_type = "source";
```

Comment:
Please use `as_screenshot_kind_from_string`, that will do the right thing here and you get an enum value to compare in the subsequent lines, instead of comparing strings.